modular_skyrat/modules/bsa_overhaul/code/bsa_computer.dm

```diff
@@ -41,15 +28,21 @@
 	. = ..()
 	ui = SStgui.try_update_ui(user, src, ui)
 	if(!ui)
-		ui = new(user, src, "BluespaceArtillery", name)
+		ui = new(user, src, "BluespaceArtillerySkyrat", name)
 		ui.open()
 
 /obj/machinery/computer/bsa_control/ui_data()
 	var/list/data = list()
-	data["ready"] = cannon ? cannon.ready : FALSE
+	var/obj/machinery/bsa/full/cannon = connected_cannon?.resolve()
 	data["connected"] = cannon
 	data["notice"] = notice
 	data["unlocked"] = GLOB.bsa_unlock
+	data["powernet_power"] = cannon?.get_available_powercap()
```

Comment:
avoid the ? checks by just doing `if(cannon)` at the start

Reply:
TGUI will bluescreen without this